CppSource/Include/xsapi/achievements_c.h

```diff
@@ -0,0 +1,235 @@
+    XSAPI_ACHIEVEMENT_PROGRESS_STATE_NOT_STARTED,
+    XSAPI_ACHIEVEMENT_PROGRESS_STATE_IN_PROGRESS
+} XSAPI_ACHIEVEMENT_PROGRESS_STATE;
+
+typedef enum XSAPI_ACHIEVEMENT_MEDIA_ASSET_TYPE
+{
+    XSAPI_ACHIEVEMENT_MEDIA_ASSET_TYPE_UNKNOWN,
+    XSAPI_ACHIEVEMENT_MEDIA_ASSET_TYPE_ICON,
+    XSAPI_ACHIEVEMENT_MEDIA_ASSET_TYPE_ART
+} XSAPI_ACHIEVEMENT_MEDIA_ASSET_TYPE;
+
+typedef enum XSAPI_ACHIEVEMENT_PARTICIPATION_TYPE
+{
+    XSAPI_ACHIEVEMENT_PARTICIPATION_TYPE_UNKNOWN,
+    XSAPI_ACHIEVEMENT_PARTICIPATION_TYPE_INDIVIDUAL,
+    XSAPI_ACHIEVEMENT_PARTICIPATION_TYPE_GROUP
+} XSAPI_ACHIEVEMENT_PARTICIPATION_TYPE;
+
+typedef enum XSAPI_ACHIEVEMENT_REWARD_TYPE
+{
+    XSAPI_ACHIEVEMENT_REWARD_TYPE_UNKNOWN,
+    XSAPI_ACHIEVEMENT_REWARD_TYPE_GAMERSCORE,
+    XSAPI_ACHIEVEMENT_REWARD_TYPE_IN_APP,
+    XSAPI_ACHIEVEMENT_REWARD_TYPE_ART
+} XSAPI_ACHIEVEMENT_REWARD_TYPE;
+
+typedef struct XSAPI_ACHIEVEMENT_TITLE_ASSOCIATION
+{
+    PCSTR name;
+    uint32_t titleId;
+
+    XSAPI_ACHIEVEMENT_TITLE_ASSOCIATION_IMPL* pImpl;
+} XSAPI_ACHIEVEMENT_TITLE_ASSOCIATION;
+
+typedef struct XSAPI_ACHIEVEMENT_REQUIREMENT
+{
+    PCSTR id;
+    PCSTR currentProgressValue;
+    PCSTR targetProgressValue;
+
+    XSAPI_ACHIEVEMENT_REQUIREMENT_IMPL* pImpl;
+} XSAPI_ACHIEVEMENT_REQUIREMENT;
+
+typedef struct XSAPI_ACHIEVEMENT_PROGRESSION
+{
+    XSAPI_ACHIEVEMENT_REQUIREMENT** requirements;
+    uint32_t requirementsCount;
+    time_t timeUnlocked;
+
+    XSAPI_ACHIEVEMENT_PROGRESSION_IMPL* pImpl;
+} XSAPI_ACHIEVEMENT_PROGRESSION;
+
+typedef struct XSAPI_ACHIEVEMENT_TIME_WINDOW
+{
+    time_t startDate;
+    time_t endDate;
+} XSAPI_ACHIEVEMENT_TIME_WINDOW;
+
+typedef struct XSAPI_ACHIEVEMENT_MEDIA_ASSET
+{
+    PCSTR name;
+    XSAPI_ACHIEVEMENT_MEDIA_ASSET_TYPE mediaAssetType;
+    PCSTR url;
+
+    XSAPI_ACHIEVEMENT_MEDIA_ASSET_IMPL* pImpl;
+} XSAPI_ACHIEVEMENT_MEDIA_ASSET;
+
+
+typedef struct XSAPI_ACHIEVEMENT_REWARD
+{
+    PCSTR name;
+    PCSTR description;
+    PCSTR value;
+    XSAPI_ACHIEVEMENT_REWARD_TYPE rewardType;
+    PCSTR valueType;
+    XSAPI_ACHIEVEMENT_MEDIA_ASSET* mediaAsset;
+
+    XSAPI_ACHIEVEMENT_REWARD_IMPL* pImpl;
+} XSAPI_ACHIEVEMENT_REWARD;
+
+typedef struct XSAPI_ACHIEVEMENT
+{
+    PCSTR id;
+    PCSTR serviceConfigurationId;
+    PCSTR name;
+    XSAPI_ACHIEVEMENT_TITLE_ASSOCIATION** titleAssociations;
+    uint32_t titleAssociationsCount;
+    XSAPI_ACHIEVEMENT_PROGRESS_STATE progressState;
+    XSAPI_ACHIEVEMENT_PROGRESSION progression;
+    XSAPI_ACHIEVEMENT_MEDIA_ASSET** mediaAssets;
+    uint32_t mediaAssetsCount;
+    PCSTR* platformsAvailableOn;
+    uint32_t platformsAvailableOnCount;
+    bool isSecret;
+    PCSTR unlockedDescription;
+    PCSTR lockedDescription;
+    PCSTR productId;
+    XSAPI_ACHIEVEMENT_TYPE type;
+    XSAPI_ACHIEVEMENT_PARTICIPATION_TYPE participationType;
+    XSAPI_ACHIEVEMENT_TIME_WINDOW* available;
+    XSAPI_ACHIEVEMENT_REWARD** rewards;
+    uint32_t rewardsCount;
+    uint64_t estimatedUnlockTime;
```

Comment:
is this not time_t?

Reply:
No it is type const std::chrono::seconds& which is a uint64